lib/uplinkc/object.go

```diff
@@ -278,6 +278,31 @@ func download(bucketRef C.BucketRef, path *C.char, cErr **C.char) C.DownloaderRe
 	})}
 }
 
+//export download_range
+// download_range returns an Object's data from specified range
+func download_range(bucketRef C.BucketRef, path *C.char, start, limit int64, cErr **C.char) C.DownloaderRef {
+	bucket, ok := universe.Get(bucketRef._handle).(*Bucket)
+	if !ok {
+		*cErr = C.CString("invalid bucket")
+		return C.DownloaderRef{}
+	}
+
+	scope := bucket.scope.child()
+
+	rc, err := bucket.DownloadRange(scope.ctx, C.GoString(path), start, limit)
+	if err != nil {
+		if !errs2.IsCanceled(err) {
```

Comment:
why not set the error if the context is cancelled?

Reply:
this is a question to the author of `download()` method (this is just adjusted copy) and it looks to be you :)

My guess would be that we don't want to interrupt flow if download was canceled on purpose.

Reply:
Haha, I was thinking about this since asking the question and I think you're right about not interrupting the flow.